apollo-adapters/src/jvmMain/kotlin/com/apollographql/apollo3/adapter/JavaTimeAdapters.kt

```diff
@@ -28,33 +28,33 @@ object JavaInstantAdapter : Adapter<Instant> {
 }
 
 /**
- * An [Adapter] that converts an ISO 8601 String like "2010-06-01T22:19:44.475Z" to/from
```

Comment:
If this string is given, an exception is thrown.

Playground
https://pl.kotl.in/S-iy9PlKg

```kotlin
val date = formatter.parse("2010-06-01T22:19:44.475Z", LocalDate::from)
val result = formatter.format(date) // exception
```

```
java.time.format.DateTimeParseException: Text '2010-06-01T22:19:44.475Z' could not be parsed: Unable to obtain LocalDate from TemporalAccessor: {InstantSeconds=1275430784, NanoOfSecond=475000000, MicroOfSecond=475000, MilliOfSecond=475},ISO of type java.time.format.Parsed
	at java.base/java.time.format.DateTimeFormatter.createError(Unknown Source)
	at java.base/java.time.format.DateTimeFormatter.parse(Unknown Source)
	at FileKt.main(File.kt:19)
	at FileKt.main(File.kt)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at executors.JavaRunnerExecutor$Companion.main(JavaRunnerExecutor.kt:27)
	at executors.JavaRunnerExecutor.main(JavaRunnerExecutor.kt)
Caused by: java.time.DateTimeException: Unable to obtain LocalDate from TemporalAccessor: {InstantSeconds=1275430784, NanoOfSecond=475000000, MicroOfSecond=475000, MilliOfSecond=475},ISO of type java.time.format.Parsed
	at java.base/java.time.LocalDate.from(Unknown Source)
	at java.base/java.time.format.Parsed.query(Unknown Source)
	... 9 more
```

Reply:
If "2010-06-01" is given, an exception is also thrown. (Before fixing the code.)

https://pl.kotl.in/24x4WU3Lo

```kotlin
import java.time.LocalDate
import java.time.format.DateTimeFormatter

private val formatter = DateTimeFormatter.ISO_INSTANT

fun main() {
    val date = formatter.parse("2010-06-01", LocalDate::from)
    val result = formatter.format(date) // exception
}
```

```
Exception in thread "main" java.time.format.DateTimeParseException: Text '2010-06-01' could not be parsed at index 10
 at java.time.format.DateTimeFormatter.parseResolved0 (:-1) 
 at java.time.format.DateTimeFormatter.parse (:-1) 
 at FileKt.main (File.kt:7) 
```